### PR TITLE
feat: replace lighting with alby-tools library

### PR DESCRIPTION
**Pull request by martinkaintas**
- Use alby-tools in order to make use of Alby's nodes and make the process of hosting the application easier
  - No need to host LND or BTC nodes anymore, we only need to host the client & server applications
- All you need now is to create a testnet account [here](https://app.regtest.getalby.com/) and place your address inside the server's `.env` file
- The user also needs an account (with bitcoin).
  - For this we can use a funded test account like the ones found [here](https://github.com/getAlby/lightning-browser-extension/wiki/Test-setup)

### CODE REVIEW

File: server/src/services/invoicing.ts [hunk 1]
```diff
@@ -1,50 +1,46 @@
...
+  async createInvoice(amount: number, description?: string): Promise<Invoice> {
+    const invoice = await this.ln.requestInvoice({ satoshi: amount, comment: description });
+    return invoice;
   }
 
-  async getInvoice(id: string): Promise<GetInvoiceResult> {
-    return getInvoice({ lnd: this.lnd, id });
+  async isPaid(invoice: Invoice): Promise<boolean> {
+    return await invoice.isPaid();
   }
```
Comment (Kalovelo) [inline server/src/services/invoicing.ts]: Seems unused

File: server/src/services/invoicing.ts [hunk 2]
```diff
@@ -1,50 +1,46 @@
...
-      getWalletInfo({ lnd: this.lnd }, (err) => {
-      if (err) {
-        console.error('Error connecting to LND', err);
-      }
-      this.connected = true;
-    });
+    this.ln = new LightningAddress(process.env.LN_ADDRESS);
+    this.ln.fetch().then(() =>
+      this.connected = true
+    );
```
Comment (Kalovelo) [inline server/src/services/invoicing.ts]: - Let's handle rejection/errors as well;
- In the future we could extract this in an async function e.g. `connect()` and remove this functionality from constructors

File: server/src/services/invoicing.ts [hunk 3]
```diff
@@ -1,50 +1,46 @@
...
+    let pollCount = 0;
+
+    const poll = async () => {
+      pollCount++;
+      const paid = await invoice.isPaid();
+      if (paid) {
+        emitter.emit('invoice_paid');
+      } else if (pollCount < maxPolls) {
+        setTimeout(poll, 200);
+      }
```
Comment (Kalovelo) [inline server/src/services/invoicing.ts]: let's add the final `else` with console.log and maybe throw an error and handle from there, you could for example, emit an error event

File: server/src/db/content.ts [hunk 4]
```diff
@@ -5,9 +5,9 @@ const posts: Record<string, Post> = {
...
 
-    Exercise can also have positive effects on mental health. It has been shown to reduce symptoms of anxiety and depression, as well as improve mood and overall well-being. This is due to the release of endorphins, which are natural feel-good chemicals in the brain. Regular exercise can also improve cognitive function, including memory and focus, and may even reduce the risk of age-related cognitive decline. Overall, the benefits of regular exercise are numerous, and it is important to incorporate physical activity into your daily routine for optimal health and well-being.`,
+    [2/2] Exercise can also have positive effects on mental health. It has been shown to reduce symptoms of anxiety and depression, as well as improve mood and overall well-being. This is due to the release of endorphins, which are natural feel-good chemicals in the brain. Regular exercise can also improve cognitive function, including memory and focus, and may even reduce the risk of age-related cognitive decline. Overall, the benefits of regular exercise are numerous, and it is important to incorporate physical activity into your daily routine for optimal health and well-being.`,
```
Comment (Kalovelo) [inline server/src/db/content.ts]: Motivation behind the paragraph counters?

**Review by Kalovelo — COMMENTED** [review]
Amazigly great work! Left a few comments, I would also suggest updating the readme file including context from this PR's description

File: server/src/services/invoicing.ts (hunk 1 above)
Comment (martinkaintas) [reply server/src/services/invoicing.ts]: Very true

File: server/src/db/content.ts (hunk 4 above)
Comment (martinkaintas) [reply server/src/db/content.ts]: No code is bug-free and the paragraph counter makes it easy to check if the server responded with the correct paragraph or if the last paragraph returned by the server is really the last paragraph
There have been instances where I had to check manually, thus I think the counters are useful, at least on the current test data

Comment (Kalovelo) [reply server/src/db/content.ts]: If there's indeed a bug, then we should resolve it as this one is a non-realistic workaround. 

But yeah, given the time constrictions, let's merge it and note it

**Review by Kalovelo — APPROVED** [review]

**Pull request merged**